core/transaction_pool/impl/transaction_pool_impl.hpp

```diff
@@ -81,19 +98,27 @@ namespace kagome::transaction_pool {
     // bans stale and invalid transactions for some amount of time
     std::unique_ptr<PoolModerator> moderator_;
 
-    // transactions with satisfied dependencies ready to leave the pool
-    std::map<common::Hash256, ReadyTransaction> ready_queue_;
+    /// All of imported transaction, contained in the pool
+    std::unordered_map<Transaction::Hash, std::shared_ptr<Transaction>>
+        imported_txs_;
+
+    /// Collection transaction with full-satisfied dependensies
+    std::unordered_map<Transaction::Hash, std::weak_ptr<Transaction>>
+        ready_txs_;
+
+    /// List of ready transaction over limit. It will be process first of all
+    std::list<std::weak_ptr<Transaction>> postponed_txs;
 
-    // transactions waiting for their dependencies to resolve
-    std::list<WaitingTransaction> waiting_queue_;
+    /// Transactions which provides specific tags
+    std::multimap<Transaction::Tag, std::weak_ptr<Transaction>>
+        tx_provides_tag_;
 
-    // tags provided by imported transactions
-    std::map<primitives::TransactionTag, boost::optional<common::Hash256>>
-        provided_tags_by_;
+    /// Transactions with resolved require of specific tags
```

Comment:
```suggestion
    /// Transactions with resolved requirement of a specific tag
```
Since it's a 'one tag -> many transactions' relation I think this would be more clear.